File: service/src/main/java/ren/imyan/service/MainActivity.java
package ren.imyan.service;

import android.annotation.SuppressLint;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {

    private Button btnStart;
    private Button btnStop;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        btnStart = findViewById(R.id.btn_start);
        btnStop = findViewById(R.id.btn_stop);
        btnStart.setOnClickListener(this);
        btnStop.setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.btn_start:
                Intent startIntent = new Intent(MainActivity.this, MyService.class);
                startService(startIntent);
                break;
            case R.id.btn_stop:
                Intent stopIntent = new Intent(MainActivity.this, MyService.class);
                stopService(stopIntent);
                break;
            default:
        }
    }
}

File: topline/src/main/java/ren/imyan/topline/ui/MainActivity.kt
package ren.imyan.topline.ui

import android.os.Bundle
import android.view.KeyEvent
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import androidx.lifecycle.ViewModelProvider
import ren.imyan.activity.ActivityCollector
import ren.imyan.topline.R
import ren.imyan.topline.core.BaseUIActivity
import ren.imyan.topline.databinding.ActivityMainBinding
import ren.imyan.topline.util.toast
import kotlin.system.exitProcess

class MainActivity : BaseUIActivity<ActivityMainBinding, MainViewModel>(R.layout.activity_main) {

    override fun initViewModel(): MainViewModel =
        ViewModelProvider(this)[MainViewModel::class.java]

    override fun initToolbar(): Pair<Toolbar, *> =
        Pair(binding.layoutToolbar.toolbar, "测试")

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        initFragmentPager()
    }

    private fun initFragmentPager() {
        binding.viewpager.apply {
            adapter = ViewPagerAdapter(
                supportFragmentManager,
                lifecycle,
                viewModel.fragmentList
            )
            isUserInputEnabled = false
            offscreenPageLimit = 3
        }
    }

    override fun onKeyDown(keyCode: Int, event: KeyEvent?): Boolean {
        var exitTime = 0L
        if (keyCode == KeyEvent.KEYCODE_BACK && event?.action == KeyEvent.ACTION_DOWN) {
            if ((System.currentTimeMillis() - exitTime) > 2000) {
                toast("再按一次退出")
                exitTime = System.currentTimeMillis()
            } else {
                ActivityCollector.finishAll()
                exitProcess(0)
            }
            return true
        }
        return super.onKeyDown(keyCode, event)
    }
}